### PR TITLE
Switch to ember [WIP]

### PR DESCRIPTION
**Pull request by martinRenou**

### CODE REVIEW

**Comment by stefanoborini** [comment]
I am not completely sold on this PR:

1. it does not seem to follow the overall ember "common patterns" that i've seen in the documentation.
2. you are mixing together the old structure and the new.
3. you are not using client side templates.
4. the extraction of the base jinja template can be done in a separate PR. Remember that we have the admin still to keep functional, but on the other hand, you need a clean slate to prepare for ember.

File: remoteappmanager/templates/home.html [hunk 1]
```diff
@@ -1,39 +1,246 @@
...
+                            </li>
+                        </ul>
+                    </li>
+                    <!-- Control Sidebar Toggle Button -->
+                </ul>
+            </div>
+        </nav>
+        </header>
+
+        <div id="ember_container"></div>
```
Comment (martinRenou) [inline remoteappmanager/templates/home.html]: This is the div element which contains the ember application

File: remoteappmanager/templates/home.html [hunk 2]
```diff
@@ -1,39 +1,246 @@
...
+
+        {% raw %}
+        <!-- Ember template for the Application list -->
+        <script type="text/x-handlebars" data-template-name="application">
```
Comment (martinRenou) [inline remoteappmanager/templates/home.html]: The main Ember application which encapsulate the sidebar containing the application_list_component associated template and the main content containing the application_view_component associated template

File: remoteappmanager/templates/home.html [hunk 3]
```diff
@@ -1,39 +1,246 @@
...
+                        application=current_application
+                    }}
+                {{/if}}
+                <!-- /.content -->
+            </div>
+            <!-- /.content-wrapper -->
+
+        </script>
+
+        <script type="text/x-handlebars" data-template-name="components/application-list">
```
Comment (martinRenou) [inline remoteappmanager/templates/home.html]: application_list_component associated template

File: remoteappmanager/templates/home.html [hunk 4]
```diff
@@ -1,39 +1,246 @@
...
+                        {{/if}}
+
+                        <span>{{app_name application.app_data}}</span>
+
+                    </a>
+                </li>
+            {{/each}}
+        </script>
+
+        <script type="text/x-handlebars" data-template-name="components/application-view">
```
Comment (martinRenou) [inline remoteappmanager/templates/home.html]: application_view_component associated template

File: remoteappmanager/static/js/home/components/application_list_component.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+    // Ember application model
+
+    const Application = Ember.Object.extend({
+        //
+    });
+
+    // Ember component for the application list
+
+    var ApplicationListComponent = Ember.Component.extend({
```
Comment (martinRenou) [inline remoteappmanager/static/js/home/components/application_list_component.js]: The Ember component representing the list of applications. It contains all the data representing the applications.

File: remoteappmanager/static/js/home/components/application_view_component.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    var Status = application_list_component.Status;
+
+    var getWindowResolution = function() {
+        var max_size = utils.max_iframe_size();
+        return max_size[0]+"x"+max_size[1];
+    };
+
+    // Ember component for the application VNC view
+
+    var ApplicationViewComponent = Ember.Component.extend({
```
Comment (martinRenou) [inline remoteappmanager/static/js/home/components/application_view_component.js]: The Ember component representing the vnc view of the current clicked application.

File: remoteappmanager/static/js/home/controller.js [hunk 7]
```diff
@@ -1,136 +1,47 @@
...
+    // Ember Controller
 
-        if (model.status[index] === models.Status.STARTING) {
-            return;
-        }
-        
-        model.status[index] = models.Status.STARTING;
-        app_view.render(false, null);
-        app_list_view.update_entry(index);
+    AppList.ApplicationController = Ember.Controller.extend({
```
Comment (martinRenou) [inline remoteappmanager/static/js/home/controller.js]: The main application controller which is the link between the application_list_component and the application_view_component. It only contains the current clicked application in the application list and keep it updated with the application view.

**Comment by martinRenou** [comment]
We decided to change our approach. We will use ember-cli to create a "subproject" webclient which will generate all the client-side code (js/css/html).

**Pull request closed without merging**